### ISSUE TITLE
🤖 [AUTO] Implement Strategy: fundamental_cross_sectional_factor_model

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#530]
## 🤖 Auto-Generated Strategy Request

**Source:** Machine Trading by Ernest P. Chan.pdf (strategy 3)
**Extracted:** 2025-12-27T05:35:24.076427

---

1.  **Strategy Name**: fundamental_cross_sectional_factor_model
2.  **Strategy Type**: Factor-based, Value (fundamental), Quarterly prediction
3.  **Timeframe**: Quarterly (63 trading days holding period, quarterly rebalancing)
4.  **Instruments**: SPX component stocks (large-cap equities)

5.  **Entry Rules**:
    *   Aggregate historical quarterly fundamental and return data from a training set across all stocks into a single dataset.
    *   Fit a linear regression model (cross-sectional) to predict next-quarter stock returns using 27 company-size independent fundamental factor loadings (listed in Table 2.1).
    *   **Long Entry**: Enter a long position in a stock on the next trading day's market close if its predicted next-quarter return is positive.
    *   **Short Entry**: Enter a short position in a stock on the next trading day's market close if its predicted next-quarter return is negative.

6.  **Exit Rules**:
    *   **Time-based**: Hold positions for one quarter (approximately 63 trading days). Rebalance all positions quarterly based on new predictions.
    *   *Take Profit Conditions*: Not specified.
    *   *Stop Loss Conditions*: Not specified.

7.  **Indicators Required**:
    *   **Fundamental Factor Loadings (27 company-size independent variables)**:
        *   **Quarterly**: CURRENTRATIO, DE (Debt to Equity Ratio), DILUTIONRATIO, PB (Price to Book Value), TBVPS (Tangible Asset Book Value per Share)
        *   **Trailing 1-year**: ASSETTURNOVER, EBITDAMARGIN, EPSGROWTH1YR, EQUITYAVG, EVEBIT (Enterprise Value over EBIT), EVEBITDA (Enterprise Value over EBITDA), GROSSMARGIN, INTERESTBURDEN, LEVERAGERATIO, NCFOGROWTH1YR, NETINCGROWTH1YR (Net Income Growth), NETMARGIN (Profit Margin), PAYOUTRATIO, PE (Price Earnings Damodaran Method), PE1, PS, PS1 (Price Sales Damodaran Method), REVENUEGROWTH1YR, ROA, ROE, ROS, TAXEFFICIENCY

8.  **Risk Management**:
    *   **Position Sizing**: Implied equal dollar weighting for each long/short position.
    *   *Maximum Risk per Trade*: Not specified.
    *   *Risk-Reward Ratio*: Not specified.

9.  **Additional Notes**:
    *   This strategy achieved a CAGR of 12.3% and a Sharpe ratio of 1.7 out-of-sample.
    *   Uses fundamental data from Sharadar's Core U.S. Fundamentals database via Quandl.com.
    *   Focuses on company-size independent factors to avoid overfitting.
    *   A model using all 112 available factors (including size-dependent ones) showed severe overfitting and negative out-of-sample performance.
    *   The R-squared of the regression fit was very low (0.015), yet the strategy was profitable.

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/fundamental_cross_sectional_factor_model.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
